src/currencyEngineETH.js

```diff
@@ -589,18 +589,27 @@ class EthereumEngine {
     if (this.addressesChecked) {
       return
     }
-    const numTokens = this.walletLocalData.enabledTokens.length
+    const activeTokens: Array<string> = []
```

Comment:
This is the perfect place to use filter:
`const activeTokens = Object.keys(this.walletLocalData.enabledTokens).filter(tokenCode => 
  this.getTokenInfo(tokenCode)
)`

Reply:
I'm avoiding functional programming in general for performance and debugability reasons.